commons-geometry-euclidean/src/test/java/org/apache/commons/geometry/euclidean/threed/hull/ConvexHull3DTest.java

```diff
@@ -0,0 +1,283 @@
+                Vector3D.of(3, 0, 0));
+        builder.append(vertices);
+        ConvexHull3D hull = builder.build();
+        assertNotNull(hull);
+        assertNull(hull.getRegion());
+        assertTrue(hull.getFacets().isEmpty());
+        assertTrue(vertices.equals(hull.getVertices()));
+    }
+
+    @Test
+    void simplex() {
+        List<Vector3D> vertices = Arrays.asList(Vector3D.ZERO, Vector3D.of(1, 0, 0), Vector3D.of(0, 1, 0),
+                Vector3D.of(0, 0, 1));
+        builder.append(vertices);
+        ConvexHull3D hull = builder.build();
+        assertNotNull(hull.getRegion());
+        assertTrue(hull.getRegion().contains(Vector3D.ZERO));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 0, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 1, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 0, 1)));
+        // The size of the simplex is finite and non zero.
+        assertTrue(TEST_PRECISION.eq(1.0 / 6.0, hull.getRegion().getSize()));
+    }
+
+    @Test
+    void simplexPlusPoint() {
+
+        List<Vector3D> vertices = Arrays.asList(Vector3D.ZERO, Vector3D.of(1, 0, 0), Vector3D.of(0, 1, 0),
+                Vector3D.of(0, 0, 1), Vector3D.of(1, 1, 1));
+        builder.append(vertices);
+        ConvexHull3D hull = builder.build();
+        assertNotNull(hull.getRegion());
+        assertTrue(hull.getRegion().contains(Vector3D.ZERO));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 0, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 1, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 0, 1)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 1, 1)));
+        assertTrue(TEST_PRECISION.eq(1.0 / 2.0, hull.getRegion().getSize()));
+
+        // Check if all facets are connected with neighbors.
+    }
+
+    @Test
+    void unitCube() {
+        List<Vector3D> vertices = Arrays.asList(Vector3D.ZERO, Vector3D.of(1, 0, 0), Vector3D.of(0, 1, 0),
+                Vector3D.of(0, 0, 1), Vector3D.of(1, 1, 0), Vector3D.of(1, 0, 1), Vector3D.of(0, 1, 1),
+                Vector3D.of(1, 1, 1));
+        builder.append(vertices);
+        ConvexHull3D hull = builder.build();
+        assertNotNull(hull.getRegion());
+        assertTrue(hull.getRegion().contains(Vector3D.ZERO));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 0, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 1, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 0, 1)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 1, 0)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 0, 1)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(0, 1, 1)));
+        assertTrue(hull.getRegion().contains(Vector3D.of(1, 1, 1)));
+        assertTrue(TEST_PRECISION.eq(1.0, hull.getRegion().getSize()));
```

Comment:
We need to assert all of the algorithm outputs in each one of these tests. This includes the
- vertices
- region
- facets
- degenerate status

Reply:
I improved the testing to assert all the outputs.